cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/inmemory/InMemoryTable.java

```diff
@@ -36,13 +38,45 @@
   private static final long NO_TX_VERSION = 0L;
 
   private Transaction tx;
+  // name length + name of the table: handy to have one cached
+  private final byte[] nameAsTxChangePrefix;
 
+  /**
+   * To be used in tests which do not need namespaces
+   */
   public InMemoryTable(String name) {
     this(name, ConflictDetection.ROW);
   }
 
+  /**
+   * To be used in tests that do not need namespaces
+   */
   public InMemoryTable(String name, ConflictDetection level) {
     super(name, level);
+    this.nameAsTxChangePrefix = Bytes.add(new byte[]{(byte) getTableName().length()}, Bytes.toBytes(getTableName()));
+  }
+
+  /**
+   * To be used in tests that need namespaces
+   */
+  public InMemoryTable(DatasetContext datasetContext, String name, CConfiguration cConf) {
+    this(datasetContext, name, ConflictDetection.ROW, cConf);
+  }
+
+  /**
+   * To be used in tests that need namespaces
+   */
+  public InMemoryTable(DatasetContext datasetContext, String name, ConflictDetection level, CConfiguration cConf) {
```

Comment:
For ``InMemoryTable``, the isolation test in ``AbstractDatasetFramework`` uses the new constructors. It also doesn't depend the name being namespaced any more.

I'm unsure about whether we should use namespaces in ``InMemoryMetricsTable``, since metrics tables always exist in the same namespace. I'll speak with @abaranau and remove the new constructors if we don't need them.